File: src/main/java/supermetroid/trackable/Trackable.java
package supermetroid.trackable;

import supermetroid.State;

import javax.swing.*;

public abstract class Trackable {
  protected JLabel label;

  public void setLable(JLabel label) {
    this.label = label;
    setDefault();
  }

  public abstract void setIcon(State state);
  public abstract void setDefault();
}

File: src/main/java/supermetroid/trackable/Trackable.kt
package supermetroid.trackable

import supermetroid.State
import javax.swing.JLabel

abstract class Trackable {
    protected var label = JLabel()

    fun initLabel(label: JLabel) {
        this.label = label
        setDefault()
    }

    abstract fun setIcon(state: State)
    abstract fun setDefault()
}
